array_schema.go

```diff
@@ -11,9 +11,54 @@ import (
 	"fmt"
 	"os"
 	"runtime"
+	"sync/atomic"
 	"unsafe"
 )
 
+type arraySchemaHandle struct {
+	// An unsafe pointer to the capiHandle holding the schema.
+	// This pointer must be accessed using atomic functions, because the MarshalJSON function
+	// replaces the native handle of an existing object.
+	// We cannot wrap atomic.Pointer[capiHandle] in a struct and pass it around because of lint warnings,
+	// and we cannot mutate the native handle of an existing capiHandle because it's impossible to atomically
+	// set the finalizer.
+	h unsafe.Pointer
+}
+
+func freeCapiArraySchema(c unsafe.Pointer) {
+	C.tiledb_array_schema_free((**C.tiledb_array_schema_t)(unsafe.Pointer(&c)))
+}
+
+func newArraySchemaCapiHandle(ptr *C.tiledb_array_schema_t) *capiHandle {
+	return newCapiHandle(unsafe.Pointer(ptr), freeCapiArraySchema)
+}
+
+func newArraySchemaHandle(ptr *C.tiledb_array_schema_t) arraySchemaHandle {
+	h := newArraySchemaCapiHandle(ptr)
+	return arraySchemaHandle{h: unsafe.Pointer(h)}
+}
+
+func (x *arraySchemaHandle) loadCapiHandle() *capiHandle {
+	return (*capiHandle)(atomic.LoadPointer(&x.h))
```

Comment:
`loadCapiHandle` and `Free` don't seem to need the arraySchemaHandle type, could they be defined in some base interface so that we don't duplicate those lines for every TileDB-Go type we want to manage? 

Kinda ditto for my last comment I guess, it seems like we could use an interface with some default implementation to save some code duplication redefining these.

Reply:
`arraySchemaHandle` is special because the `*capiHandle` can be mutated, which means that all accesses to `h` must be atomic. The other handles are immutable and don't need the extra level of indirection in `loadCapiHandle`, they can just read it.